pkg/process/metadata/workloadmeta/extractor.go

```diff
@@ -16,11 +16,12 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
-// ProcessEntity is a placeholder workloadmeta.ProcessEntity.
-// It does not contain all the fields that the final entity will contain.
+// ProcessEntity represents a process exposed by the WorkloadMeta extractor
```

Comment:
The current data path is `Process Entity` (Extractor cache) -> `EventSet` (gRPC connection) -> `Process` (Workloadmeta).

The Process fields defined in the Workloadmeta are not following the same format of the fields from a Process collected with procutil, e.g. creationTime is using `time.Time` instead of `int64`.
https://github.com/DataDog/datadog-agent/blob/c19c9fb597f7ced008a8421a19fb846ad27e0267/pkg/workloadmeta/types.go#L872-L880

To prevent an additional conversion in the Extractor and given that a conversion is already needed to send the message through gRPC, I've opted to keep the cache definition isolated from the workloadmeta one.

This also keeps the Extractor and the Workloadmeta uncoupled so the former can eventually expose process data to other consumers without any restriction.

Reply:
I'm fine with this change, however it might be better to use `procutil.Process` in this case to avoid any conversions, WDYT?